src/base/bittorrent/tracker.cpp

```diff
@@ -271,52 +275,95 @@ void Tracker::processAnnounceRequest()
     // ip address
     announceReq.socketAddress = m_env.clientAddress;
     announceReq.claimedAddress = queryParams.value(ANNOUNCE_REQUEST_IP);
+    // request port
+    announceReq.requestPort =  m_env.clientPort;
 
     // Enforce using IPv4 if address is indeed IPv4 or if it is an IPv4-mapped IPv6 address
     bool ok = false;
     const qint32 decimalIPv4 = announceReq.socketAddress.toIPv4Address(&ok);
-    if (ok)
+    if (ok) {
         announceReq.socketAddress = QHostAddress(decimalIPv4);
+        LogMsg(tr(R"(Embedded Tracker: Announce request from non- plain-IPv6 address %1.)"
```

Comment:
You call LogMsg here and at exception handler, what a mess...
At this point this PR is beyond my comprehension, sorry.

Reply:
This is not constructive criticism. Do you mean I should remove this logging statement specifically, or should I remove the `LogMsg` call from the `HTTPError`, or both? If so, why?

Reply:
Sorry, all of this is over my head, I couldn't even give a proper response...

Reply:
@Chocobo1

I mean, am I doing something really bad or what? I just wanted a bit more logging overall, am I doing it in the wrong places in general?

In this specific place and the `HTTPError` exception handling, the behavior doesn't change at all, the only difference is the logging statements.